File: truss-compiler/src/main/java/truss/compiler/syntax/SyntaxLibrary.java
package truss.compiler.syntax;

import org.apache.commons.lang.Validate;
import truss.compiler.support.ImmutableArray;

public class SyntaxLibrary {
    private final ImmutableArray<CompilationUnitSyntax> compilationUnits;

    public SyntaxLibrary(ImmutableArray<CompilationUnitSyntax> compilationUnits) {
        Validate.notNull(compilationUnits, "compilationUnits");

        this.compilationUnits = compilationUnits;
    }

    public ImmutableArray<CompilationUnitSyntax> getCompilationUnits() {
        return compilationUnits;
    }
}

File: Truss.Compiler/Syntax/SyntaxLibrary.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Truss.Compiler.Support;

namespace Truss.Compiler.Syntax {
    public class SyntaxLibrary {
        public SyntaxLibrary(ImmutableArray<CompilationUnitSyntax> compilationUnits) {
            if (compilationUnits == null) {
                throw new ArgumentNullException("compilationUnits");
            }

            CompilationUnits = compilationUnits;
        }

        public ImmutableArray<CompilationUnitSyntax> CompilationUnits { get; private set; }
    }
}
